### PR TITLE
IOTEVENT-669

### PR DESCRIPTION
**Pull request by edwardsoo-ss**
get log level from request headers and propagate using MDC
added dynamic threshold logger filter
added DW logger filter factories for dynamic and static threshold filter

@charliek @Bijnagte

### CODE REVIEW

File: dropwizard-logging/src/main/java/smartthings/dw/logging/LoggingContext.java [hunk 1]
```diff
@@ -3,8 +3,10 @@
 import org.slf4j.MDC;
 
 public class LoggingContext {
-	public final static String CORRELATION_ID_HEADER = "X-ST-CORRELATION";
-	protected final static String LOGGING_ID = "loggingId";
+    public final static String CORRELATION_ID_HEADER = "X-ST-CORRELATION";
+    public final static String LOG_LEVEL_HEADER = "X-ST-LOG-LEVEL";
+	public final static String LOGGING_ID = "loggingId";
```
Comment (charliek) [inline dropwizard-logging/src/main/java/smartthings/dw/logging/LoggingContext.java]: Indenting off here?

**Comment by charliek** [comment]
Could you paste in an example logging section example for this? Since DW has the logging config more integrated into the framework than the ratpack stuff we do does.

File: dropwizard-logging/src/main/java/smartthings/dw/logging/filter/DynamicThresholdLoggerFilterFactory.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.collect.Lists;
+import io.dropwizard.logging.filter.FilterFactory;
+
+import java.util.List;
+
+@JsonTypeName("dynamic")
+public class DynamicThresholdLoggerFilterFactory implements FilterFactory<LoggingEvent> {
+
+    @JsonProperty
```
Comment (charliek) [inline dropwizard-logging/src/main/java/smartthings/dw/logging/filter/DynamicThresholdLoggerFilterFactory.java]: Should these be non-nullable so they catch issues that might be in the config?  Should we default anything here to reasonable values?

Comment (edwardsoo-ss) [inline dropwizard-logging/src/main/java/smartthings/dw/logging/filter/DynamicThresholdLoggerFilterFactory.java]: the filter itself has defaults (I will add a comment).
if the dw config has invalid values `ch.qos.logback.classic.Level#toLevel(java.lang.String)` and `ch.qos.logback.core.spi.FilterReply#valueOf` should throw

Comment (charliek) [inline dropwizard-logging/src/main/java/smartthings/dw/logging/filter/DynamicThresholdLoggerFilterFactory.java]: ok

**Comment by edwardsoo-ss** [comment]
added a README to dropwizard-logging with a log filter usage example

**Pull request merged**